### ISSUE TITLE
Spellchecking not working, unable to find out why

### ISSUE THREAD

**Issue by Schinnerl-at** [IsmaelMartinez/teams-for-linux#1282]
**Describe the bug**
Note: English is not my first language.

I have to use Teams in a somewhat restricted network, that means only some domains are accessible via a proxy. And now I have the problem that no spellchecking seams to work. 

**To Reproduce**

Setup a proxy that only allows:
.microsoft.com, .microsoft.de, login.live.com, login.microsoftonline.com, 
.gstatic.com, .googleapis.com, .googlecode.com, .googleusercontent.com, apis.google.com,
code.jquery.com, consent.google.at, consent.google.com
.github.com, .github.io, .githubusercontent.com,
.mozaws.net, .mozilla.com, .mozilla.net, .mozilla.org,
.npmjs.org

Note: that is not everything that is allowed, but that should be everything that teams for linux needs, if there is something missing, than I found the bug.

1. Start teams-for-linux with following arguments:
   --webDebug true --appLogLevels "error,warn,info,debug" --proxyServer "<proxy server>" --spellCheckerLanguages "en-US,de,de-DE"
3. Click on any chat
4. input something that is spelled worng.
5. no "redline" under the word, and a context menu does not show suggestions.
6. webDebuger and log output does not give any hints on what is wrong

**Expected behavior**
Nice would be a working spellchecking, but what I need more is a way to find out why it is not working, there is no debug log that says that something does not work. Also the webDebuger has no 403 or something like that.

**Desktop (please complete the following information):**
 - OS: debian 12.5
 - Installation package deb
 - Version 1.5.3
 - Installed hunspell packages: 
   - dictionaries-common/stable,now 1.29.5 all [installed,automatic]
   - hunspell-de-at/stable,now 20161207-11 all [installed]
   - hunspell-de-de/stable,now 20161207-11 all [installed]
   - hunspell-en-us/stable,now 1:2020.12.07-2 all [installed,automatic]
   - libhunspell-1.7-0/stable,now 1.7.1-1 amd64 [installed,automatic]

**Debug**
```
[DEBUG] configPath:,/home/schich/.config/teams-for-linux
[DEBUG] configFile:,[object Object]
[INFO] Running under Wayland, switching to PipeWire...
[DEBUG] Custom background service url is 'http://localhost'
[DEBUG] processArgs:,/opt/teams-for-linux/teams-for-linux,--webDebug,true,--appLogLevels,error,warn,info,debug
[DEBUG] Testing network using net.request() for https://teams.microsoft.com/
[DEBUG] Language codes en-US,de set for spellchecker
[7393:0606/094114.234308:ERROR:browser_main_loop.cc(278)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
[7393:0606/094114.234926:ERROR:browser_main_loop.cc(278)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
[7393:0606/094114.235068:ERROR:browser_main_loop.cc(278)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
[7393:0606/094114.235140:ERROR:browser_main_loop.cc(278)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
[7393:0606/094114.237315:ERROR:browser_main_loop.cc(278)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
[DEBUG] Network test successful with method https
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[7393:0606/094114.738356:ERROR:object_proxy.cc(576)] Failed to call method: org.freedesktop.ScreenSaver.GetActive: object_path= /org/freedesktop/ScreenSaver: org.freedesktop.DBus.Error.NotSupported: This method is not implemented
[DEBUG] did-frame-finish-load,[object Object],true
[DEBUG] did-finish-load
[DEBUG] did-frame-finish-load,[object Object],true
[DEBUG] did-finish-load
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[7393:0606/094115.836979:ERROR:CONSOLE(1)] "Request Autofill.enable failed. {"code":-32601,"message":"'Autofill.enable' wasn't found"}", source: devtools://devtools/bundled/core/protocol_client/protocol_client.js (1)
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] did-frame-finish-load,[object Object],true
[DEBUG] did-finish-load
[DEBUG] Badge count set to '0'
[DEBUG] Badge count set to '0'
[DEBUG] Badge count set to '0'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] did-frame-finish-load,[object Object],false
[7393:0606/094132.861756:ERROR:CONSOLE(1)] "Request Autofill.enable failed. {"code":-32601,"message":"'Autofill.enable' wasn't found"}", source: devtools://devtools/bundled/core/protocol_client/protocol_client.js (1)
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[7459:0606/094139.661560:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
[7459:0606/094145.615886:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!
[7459:0606/094145.626932:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 3 times!
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] Badge count set to '0'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] Language codes en-US,de,de-DE set for spellchecker
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
^C[DEBUG] before-quit
[DEBUG] window close
[DEBUG] window closed
[DEBUG] will-quit
[DEBUG] quit
```

**Additional context**
config.json:
```
{
	"proxyServer": "<proxy server>",
	"customCSSName": "compactDark",
	"closeAppOnCross": true,
	"defaultURLHandler": "/home/schich/.config/teams-for-linux/opennewtab.sh",
	"optInTeamsV2": true,
	"notificationMethod": "electron",
	"spellCheckerLanguages": [
		"en-US",
		"de",
		"de-DE"
	]
}
```

I hope that is detailed enough, and thanks for the great teams client.

**Comment by IsmaelMartinez** [comment]
Hi, I don't understand why this would not work, but it could well be because of the customCSSName used. It might be messing up with the spellchecker. 

I would also not provide more than 1 "de" options, but that I suspect wouldn't be a problem.

Do you get a "Writing Language" option if you right click on a word?

<img width="296" alt="image" src="https://github.com/IsmaelMartinez/teams-for-linux/assets/430992/ded20d57-3c6d-4aae-b5fe-3b31eae192e8">

**Comment by IsmaelMartinez** [comment]
Closing due inactivity. Happy to reopen if/when more information is provided

**Comment by Schinnerl-at** [comment]
Sorry, I only have the problem on my work machine and I was not on it for some time.

Here the screenshot, what else can I do to search for a solution?
![Screenshot from 2024-06-25 08-48-40](https://github.com/IsmaelMartinez/teams-for-linux/assets/26482123/4d255cd4-c557-4106-a855-e5c591606af2)

**Comment by IsmaelMartinez** [comment]
have you tried without the following config options:

```
{
	"proxyServer": "<proxy server>",
	"closeAppOnCross": true,
	"defaultURLHandler": "/home/schich/.config/teams-for-linux/opennewtab.sh",
	"optInTeamsV2": true,
	"notificationMethod": "electron",
	"spellCheckerLanguages": [
		"en-US",
		"de"
	]
}
```

Basically I can see in the logs that de-DE is not been pick up 
```
[DEBUG] Language codes en-US,de set for spellchecker
```

But what I am more suspicious is on the `"customCSSName": "compactDark",` that might be altering the view.

It can also be that spellchecker is not working for teams-v2 but lets 1st see if those changes do anything with it.

**Comment by Schinnerl-at** [comment]
I have now tested this configs:
```
{
	"proxyServer": "172.25.5.60:8080",
	"closeAppOnCross": true,
	"defaultURLHandler": "/home/schich/.config/teams-for-linux/opennewtab.sh",
	"optInTeamsV2": true,
	"notificationMethod": "electron",
	"spellCheckerLanguages": [
		"de",
		"en"
	]
}
```
and:
```
{
	"proxyServer": "172.25.5.60:8080",
	"closeAppOnCross": true,
	"defaultURLHandler": "/home/schich/.config/teams-for-linux/opennewtab.sh",
	"optInTeamsV2": false,
	"notificationMethod": "electron",
	"spellCheckerLanguages": [
		"de",
		"en"
	]
}
```

here the log from the second run `"optInTeamsV2": false`:
```
[DEBUG] configPath:,/home/schich/.config/teams-for-linux
[DEBUG] configFile:,[object Object]
[INFO] Running under Wayland, switching to PipeWire...
[DEBUG] Custom background service url is 'http://localhost'
[DEBUG] processArgs:,/opt/teams-for-linux/teams-for-linux,--webDebug,true,--appLogLevels,error,warn,info,debug
[DEBUG] testing RegExp processArgs false
[DEBUG] testing RegExp processArgs false
[DEBUG] testing RegExp processArgs false
[DEBUG] testing RegExp processArgs false
[DEBUG] testing RegExp processArgs false
[DEBUG] Testing network using net.request() for https://teams.microsoft.com/
[DEBUG] Language codes de,en set for spellchecker
[28959:0702/090907.789916:ERROR:browser_main_loop.cc(278)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
[28959:0702/090907.790327:ERROR:browser_main_loop.cc(278)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
[28959:0702/090907.790466:ERROR:browser_main_loop.cc(278)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
[28959:0702/090907.790550:ERROR:browser_main_loop.cc(278)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
[28959:0702/090907.792303:ERROR:browser_main_loop.cc(278)] Gtk: gtk_widget_add_accelerator: assertion 'GTK_IS_ACCEL_GROUP (accel_group)' failed
[DEBUG] Network test successful with method https
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[28959:0702/090908.546023:ERROR:object_proxy.cc(576)] Failed to call method: org.freedesktop.ScreenSaver.GetActive: object_path= /org/freedesktop/ScreenSaver: org.freedesktop.DBus.Error.NotSupported: This method is not implemented
[28959:0702/090909.132521:ERROR:CONSOLE(1)] "Request Autofill.enable failed. {"code":-32601,"message":"'Autofill.enable' wasn't found"}", source: devtools://devtools/bundled/core/protocol_client/protocol_client.js (1)
[DEBUG] did-frame-finish-load,[object Object],true
[DEBUG] did-finish-load
[DEBUG] Badge count set to '0'
[DEBUG] Badge count set to '0'
[DEBUG] Badge count set to '0'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] Badge count set to '5'
[DEBUG] Badge count set to '7'
[DEBUG] did-frame-finish-load,[object Object],false
[28959:0702/090925.575902:ERROR:CONSOLE(1)] "Request Autofill.enable failed. {"code":-32601,"message":"'Autofill.enable' wasn't found"}", source: devtools://devtools/bundled/core/protocol_client/protocol_client.js (1)
[29026:0702/090928.668607:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[29026:0702/090933.560586:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!
[29026:0702/090937.901387:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 3 times!
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] GetSystemIdleState => IdleTimeout: 300s, IdleTimeoutPollInterval: 10s, ActiveCheckPollInterval: 2s, IdleTime: 0s, IdleState: 'active'
[DEBUG] window close
[DEBUG] window closed
[DEBUG] before-quit
[DEBUG] will-quit
[DEBUG] quit
```

I am only once or twice a week on that PC, so that ticket will be really slow, also I don't think that has a high priority.

How can I help you?

**Comment by IsmaelMartinez** [comment]
Can you put the logs on the UI? Just clicking on that Debug. 
![image](https://github.com/IsmaelMartinez/teams-for-linux/assets/430992/ca31d5dd-9315-4e5d-8fb5-a86009078c0e)

The go to Console 
![image](https://github.com/IsmaelMartinez/teams-for-linux/assets/430992/d62c4c1a-16ae-42bb-a336-131fb4382032)

Clear the console (the second icon)
![image](https://github.com/IsmaelMartinez/teams-for-linux/assets/430992/6e6d3492-645b-40c3-b19d-78a6517134aa)

And type some text that is wrong and see if it spits anything.

Worth checking if this README information is relevant anymore https://github.com/IsmaelMartinez/teams-for-linux/blob/develop/KNOWN_ISSUES.md#spellchecker-not-working . Spellchecker is an experimental feature that works in most cases, but it has a few caveats.

**Comment by IsmaelMartinez** [comment]
Ok, this might be now fixed in [pre-release 1.9.1](https://github.com/IsmaelMartinez/teams-for-linux/releases/tag/v1.9.1) using the config option `disableGpu` that was missing an extra electron command. See https://github.com/IsmaelMartinez/teams-for-linux/issues/1369 for more info. I saw some errors in your logs that might point to that. I assume this fixes it but re-open if it doesn't. Hope helps!

**Comment by Schinnerl-at** [comment]
Today I noticed in 2.0.16 that this problem does not exist any more. I have no idea why.

Now I have no excuse to have bad typos, I don't know how I feel about it.

Thanks for your work and this project.
